### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by PrangonGhose**
### Changes made in this PR
- Used [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to store, retrieve, and delete books.
- Fetch book list from bookstore API on page load.
- Refactored _add book_ and _remove book_ features to persist changes in the server.
- _Add book_ and _Remove book_ works in the same way.
- No styling is done.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,14 +1,29 @@
-import { ADD_BOOK, REMOVE_BOOK } from '../constants';
+import { ADD_BOOK, ALL_BOOKS, REMOVE_BOOK } from '../constants';
+import { postBooks, deleteBook, getBooks } from '../../components/connectAPI';
 
-export const AddBook = (data) => ({
-  type: ADD_BOOK,
-  data,
-});
+export const AddBook = (data) => async (dispatch) => {
```
Comment (topeogunleye) [inline src/redux/books/books.js]: - [ ] As required, I kindly suggest that you make use of createAsyncThunk for thunk action creators

File: src/components/connectAPI.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Ze0tRMOWnvW25f5ddSv1/books/';
+
+export const getBooks = async () => {
```
Comment (topeogunleye) [inline src/components/connectAPI.js]: - Kindly use [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) for making API calls.

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @PrangonGhose, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrangonGhose** [comment]
### Changes made after first review
- Use `createAsyncThunk` along with `fetch` to make API calls

**Review by franclobo — APPROVED** [review]
Hi @PrangonGhose,

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- All changes required completed 💯 
- Good use of createAsyncThunk ✔️
- Add and remove functions work well ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**